app/nlg/country.py

```diff
@@ -84,28 +86,23 @@ def render(cnc, lexeme, entity):
 
 	region_nps  = []
 	region_advs = []
-	preposition = w.of_1_Prep if cnc.name in ["ParseSpa", "ParseFre"] else w.in_1_Prep
 	if "Q23522" in location_qids:  # on the balkans instead of in the balkans
 		region_advs.append(mkAdv(w.on_1_Prep,mkNP(w.balkans_2_LN)))
 	if not region_advs or entity["id"] == "Q43":
 		for qid in regions: # the Caribbean, Melanesia, etc.
 			if qid in part_of_qids or qid in location_qids:
 				lex_fun = cnc.get_lex_fun(qid)
 				if lex_fun:
-					if lex_fun in [w.middle_east_LN, w.caucasus_LN, w.caribbean_LN]:
-						region_nps.append(mkNP(lex_fun))
-					else:
-						region_nps.append(w.PlainLN(lex_fun))
-			
+					region_nps.append(mkNP(lex_fun))
 	if region_nps:
-		region_advs.append(mkAdv(preposition,mkNP(w.and_Conj,region_nps)))
+		region_advs.append(mkAdv(w.in_1_Prep,mkNP(w.and_Conj,region_nps)))
 	if region_advs:
 		cn = mkCN(cn,mkAdv(w.and_Conj,region_advs))
 	if not has_adjective and not region_advs:
 		# add the continent if stated
 		for continent in cnc.get_lexemes("P30",entity,qual=False):
-			region_nps.append(w.PlainLN(continent))
-		cn = mkCN(cn,mkAdv(preposition,mkNP(w.and_Conj,region_nps)))
+			region_nps.append(mkNP(continent))
```

Comment:
Does mkNP produce something different from PlainLN in Russian? If it does then the change should be made conditional on the language